addons/osio-addon/src/main/java/io/fabric8/launcher/osio/identity/OsioIdentityProvider.java

```diff
@@ -2,37 +2,38 @@
 
 import java.util.Optional;
 
-import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
 
 import io.fabric8.launcher.base.identity.Identity;
 import io.fabric8.launcher.base.identity.IdentityFactory;
-import io.fabric8.launcher.base.identity.TokenIdentity;
 import io.fabric8.launcher.core.spi.Application;
 import io.fabric8.launcher.core.spi.IdentityProvider;
 import io.fabric8.launcher.osio.client.OsioApiClient;
-import io.fabric8.launcher.osio.client.OsioApiClientImpl;
+import io.fabric8.launcher.osio.client.Tenant;
 
-import static io.fabric8.launcher.base.identity.IdentityFactory.createFromToken;
-import static io.fabric8.launcher.base.identity.IdentityHelper.removeBearerPrefix;
 import static io.fabric8.launcher.core.spi.Application.ApplicationType.OSIO;
 import static io.fabric8.launcher.osio.OsioConfigs.ExternalServices.getGithubServiceName;
 
 
-@ApplicationScoped
+@Dependent
 @Application(OSIO)
 public class OsioIdentityProvider implements IdentityProvider {
 
+    @Inject
+    private OsioApiClient osioApiClient;
+
+    @Inject
+    private Tenant tenant;
 
     @Override
-    public Optional<Identity> getIdentity(final String service, final String authorizationHeader) {
-        final TokenIdentity osioIdentity = createFromToken(removeBearerPrefix(authorizationHeader));
-        final OsioApiClient osioApiClient = new OsioApiClientImpl(osioIdentity);
+    public Optional<Identity> getIdentity(final String service) {
         switch (service) {
             case IdentityProvider.ServiceType.GITHUB:
                 return osioApiClient.getTokenForService(getGithubServiceName())
                         .map(IdentityFactory::createFromToken);
             case IdentityProvider.ServiceType.OPENSHIFT:
-                return Optional.of(osioIdentity);
+                return Optional.of(osioApiClient.getTenant().getIdentity());
```

Comment:
Why not return tenant.getIdentity() here?

Reply:
mistake :)